Library/Homebrew/formula_installer.rb

```diff
@@ -344,7 +344,8 @@ def expand_dependencies(deps)
     inherited_options = {}
 
     expanded_deps = Dependency.expand(formula, deps) do |dependent, dep|
-      options = inherited_options[dep.name] = inherited_options_for(dep)
+      inherited_options[dep.name] ||= Options.new
+      options = inherited_options[dep.name] += inherited_options_for(dep)
```

Comment:
Yeah I'd been wondering why there was a separate copy if it was only then being passed by value in `dep.satisfied?(options)`.